src/libraries/Microsoft.PowerFx.Core/Public/Types/TableType.cs

```diff
@@ -31,10 +32,20 @@ public override void Visit(ITypeVistor vistor)
             vistor.Visit(this);
         }
 
-        public TableType Add(NamedFormulaType field)
+        public TableType Add(NamedFormulaType field, string optionalDisplayName = null)
```

Comment:
>Add [](http://example.com/codeflow?start=25&length=3)

We're mixing mutable and immutable data types:

The fact that this returns a new TableType implies that it's immutable. 
But both TableTypes actually share the same displayNameProvider, and that's mutated.

Reply:
Resolved. DisplayNameProvider now is more immutable w.r.t FormulaType constructors